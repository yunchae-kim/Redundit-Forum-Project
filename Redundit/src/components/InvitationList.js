import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import groupIcon from "../test_data/groupIcon.jpg";
import { getAllUsersInGroup } from "../utility/axiosUser";
import { deleteMember, putMember } from "../utility/axiosMember";
import { getGroupById } from "../utility/axiosGroup";
import { addNotification } from "../utility/axiosNotification";

const InvitationList = ({ groupId }) => {
  const [members, setMembers] = useState([]);
  const [group, setGroup] = useState({});
  useEffect(() => {
    getGroupById(groupId).then((groupFetched) => {
      setGroup(groupFetched);
    });
  }, []);

  useEffect(() => {
    getAllUsersInGroup(groupId).then((membersFetched) => {
      setMembers(membersFetched.filter((m) => m.pending === 1));
    });
  }, [members]);

  const deleteInvite = (iduser) => {
    deleteMember(groupId, iduser).then(() => {
      addNotification(iduser, `You are denied to join ${group.name}`).then(() => {
        setMembers([]);
      });
    });
  };

  const approveInvite = (iduser) => {
    putMember(iduser, groupId, false, false).then(() => {
      addNotification(iduser, `You are approved to join ${group.name}`).then(() => {
        setMembers([]);
      });
    });
  };

  const invitedMembers = members.map((m) => (
    <div key={m.iduser}>
      <div className="ML-entry">
        <img className="ML-userIcon" src={groupIcon} alt="" />
        <Link to={{ pathname: `/user/${m.iduser}`, state: { userId: m.iduser } }} className="ML-entry-name" key={m.iduser}><span>{m.username}</span></Link>
        <button type="button" onClick={() => deleteInvite(m.iduser)}>Delete</button>
        <button type="button" onClick={() => approveInvite(m.iduser)}>Approve</button>
      </div>
      <hr className="ML-hr" />
    </div>
  ));

  return (
    <div>
      <div>
        {invitedMembers}
      </div>
    </div>
  );
};

export default InvitationList;
