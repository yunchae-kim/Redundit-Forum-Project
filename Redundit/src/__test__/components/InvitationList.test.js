import {
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import InvitationList from "../../components/InvitationList";

test("test should InvitationList match snapshot", async () => {
  const tree = renderer.create(
    <BrowserRouter>
      <InvitationList groupId={1} />
    </BrowserRouter>,
  ).toJSON();
  await waitFor(async () => {
    expect(tree).toMatchSnapshot();
  });
});
