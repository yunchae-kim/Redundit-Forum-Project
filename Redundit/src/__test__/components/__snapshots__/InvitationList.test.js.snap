// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should InvitationList match snapshot 1`] = `
<div>
  <div />
</div>
`;
